### ISSUE TITLE
"zola  serve" doesn't detect any change

### ISSUE THREAD

**Issue by arkydon** [Keats/gutenberg#2551]
# Bug Report

## Environment

Zola version: 0.19.1, 0.19.0
Windows 11 22H2, Build version: 22621.3737

## Expected Behavior
Using "zola serve" command, it should detect changes to a website and reflect it.

## Current Behavior
In both 0.19.0 and 0.19.1 it can't detect any changes anymore, so I have to rerun "zola serve" every time I make any changes in my website code. In the terminal it's always at the initial "Listening for changes".

## Step to reproduce
1. Run command "zola serve"
2. Make any change to templates or sass or content.

# Note
I tried it in Windows Terminal, Powershell and the old command prompt. I tried running as admin too.

**Comment by Keats** [comment]
We've fixed it for Linux but for Windows I'd need someone with a Windows machine to check what's happening. See https://github.com/getzola/zola/pull/2542 for the linux PR

**Comment by knzai** [comment]
I'm seeing this in Mac as well (no changes result in a reload, even to templates or pages or anything), but maybe the linux fix will also fix that.

**Comment by Keats** [comment]
Have you tried 0.19.1? It has always worked for me on Mac

**Comment by arkydon** [comment]
> Have you tried 0.19.1? It has always worked for me on Mac

Yes. Both 0.19.0 and 0.19.1 have this issue.

**Comment by myfonj** [comment]
Tried [`zola 0.19.1`](https://github.com/getzola/zola/releases/tag/v0.19.1) on `Windows 10 Version 10.0.19045 Build 19045` (With latest Firefox and Edge) and watcher still does not seem to catch any change `:(`. 

According to devtools the websocket is being established (`ws://127.0.0.1:1024/livereload`) but never sends any message.

Most recent release with working live reload on my system seems to be [`0.18.0`](https://github.com/getzola/zola/releases/tag/v0.18.0).

**Comment by knzai** [comment]
I was already on 0.19.1, but it started working with no changes, so idek. I've historically seen mac's do weird things with reloading dev server file detection before, so probably not anything on your end. Just following up to confirm I am *not* seeing this behavior on Mac, so ignore my earlier comment I guess.

Maybe a permissioning hiccup or something on one of my installs.

**Comment by Snowdrama** [comment]
I also ran into this issue, I just installed it yesterday. Not that I'm 100% sure if it's related but it's showing the path and I'm curious if it's maybe related to the issue. 

I'm using 0.19.1
```
D:\Workspace\Rust\snowdrama_net_zola>zola.exe -V
zola 0.19.1
```

For me it's not catching any changes, templates, content anything. 

My path: 
`D:\Workspace\Rust\snowdrama_dot_net_zola\`

But it's listing the path as:
`\\?\D:\Workspace\Rust\snowdrama_net_zola\`

The full output:
```
D:\Workspace\Rust\snowdrama_net_zola>zola.exe serve 
Building site...
Checking all internal links with anchors.
> Successfully checked 10 internal link(s) with anchors.
-> Creating 6 pages (0 orphan) and 12 sections
Done in 1.3s.

Web server is available at http://127.0.0.1:1025 (bound to 127.0.0.1:1025)

Listening for changes in \\?\D:\Workspace\Rust\snowdrama_net_zola\{config.toml,content,sass,static,templates,themes}
Press Ctrl+C to stop
```

Edit: Nevermind, I checked `0.18.0` and path is the same in and it does work, so it's not related to the path.

**Comment by iamorphen** [comment]
I happened across this issue when browsing for something else. I set up a build environment in Windows 10, built c20b336cd9e97701dff6434039627c1293b03687, and can reproduce the issue. I'm not currently working on this but can go deeper when I have some free time next after #2441.

**Comment by chipotle** [comment]
> I'm seeing this in Mac as well (no changes result in a reload, even to templates or pages or anything), but maybe the linux fix will also fix that.

I'm also seeing this with Zola 0.19.1 currently.

**Comment by Keats** [comment]
Can you try https://github.com/getzola/zola/pull/2575 ?

**Comment by Snowdrama** [comment]
@Keats That seems to work for me.

Version shows me using 0.19.1
```powershell
PS D:\Workspace\Rust\snowdrama_dot_net_zola> zola.exe -V    
zola 0.19.1
```
And this does correctly reload when an update is made
```powershell
PS D:\Workspace\Rust\SnowdramaDotNetZola> zola.exe serve 
Building site...
Checking all internal links with anchors.
> Successfully checked 16 internal link(s) with anchors.
-> Creating 8 pages (1 orphan) and 12 sections
Done in 3.0s.

Web server is available at http://127.0.0.1:1111 (bound to 127.0.0.1:1111)

Listening for changes in \\?\D:\Workspace\Rust\snowdrama_dot_net_zola\{config.toml,content,sass,static,templates,themes}
Press Ctrl+C to stop

Change detected @ 2024-07-18 12:22:07
-> Content changed \\?\D:\Workspace\Rust\snowdrama_dot_net_zola\content\devlog\2024\07\test.md
Checking all internal links with anchors.
> Successfully checked 16 internal link(s) with anchors.
-> Creating 8 pages (1 orphan) and 12 sections
Done in 329ms.
```

**Comment by Keats** [comment]
Nice, now let's see if it works for @chipotle as well. Which editor are you using?

**Comment by opeik** [comment]
I'm using zola 0.19.1 on macOS 14.1.1, `zola serve` refuses to rebuild when changes occur in my theme. I suspect it's because I've symlinked it.

**Comment by iamorphen** [comment]
@opeik , I can reproduce that on macOS. Did your workflow used to work for you using zola 0.18? I took a look at the [`watch`](https://docs.rs/notify/6.1.1/notify/trait.Watcher.html#tymethod.watch) options for what we're using and didn't see anything about symlinks. I also just debugged and didn't see any captured events when modifying files through a symlink even before event filtering.

**Comment by opeik** [comment]
@iamorphen I don't think it's ever worked.

**Comment by iamorphen** [comment]
@opeik , thanks for confirming. I couldn't find any special mention of symlinks in the older version of notify-rs (4.x) we used, either. I just confirmed that your use case works on Linux, so this behavior seems to be a quirk between macOS's FSEvents and the notify/debouncer crate.

**Comment by Keavon** [comment]
@Keats I've tested that PR and it fixes it when I save files in VS Code on Windows 10. I hope that can be merged and released ASAP. Thanks.

**Comment by myfonj** [comment]
Confirming that `zola serve` with [zola-v0.19.2-x86_64-pc-windows-msvc.zip](https://github.com/getzola/zola/releases/download/v0.19.2/zola-v0.19.2-x86_64-pc-windows-msvc.zip) does the live reload upon changes on Windows 10 again 🎉, thanks for the fix!

As mentioned in the #2575, while `Ctrl+C` terminated it writes `Encountered an error: uncategorized error` before exiting, but it does not seem like it prevents anything.

**Comment by Keats** [comment]
Can you open another issue for that? I haven't seen it on Mac so it would helpful to have someone on Windows debug it
